### DISCUSSION TITLE
MFA

### DISCUSSION THREAD

**Discussion by starredev** [pocketbase/pocketbase#5636] [Q&A] [answered]
How does MFA work actually? I'm getting MFA Id but what can I do with that specific ID? 
I'm quite new into this, just asking :)

**Comment by ganigeorgiev** [comment, marked as answer]
It will be documented with the final release.

Currently the MFA requires the user to authenticated with 2 different auth methods - password, OTP, OAuth2 (the order doesn't matter).

You can find an example of password + OTP in the [JS SDK](https://github.com/pocketbase/js-sdk/releases/tag/v0.22.0-rc) and [Dart SDK](https://github.com/pocketbase/dart-sdk/blob/develop/CHANGELOG.md) changelogs.

The general flow is:

1. User authenticates with auth method A.
2. On success return 401 with an `{"mfaId": "..."}` body (this is stored in the `_mfas` system collections).
3. User authenticate with auth method B and adds the `mfaId` from step 2 either as query or body param, both should work
4. On success a regular auth response is returned, aka. token + auth record data.

Note that I still haven't decided whether to fully lock the `_mfas` and `_otps` collection or to allow list/view access for the authenticated issuers, so keep that in mind if you are planning to allow users to see their issued codes.
